app/controllers/instruments_dashboard_controller.rb

```diff
@@ -26,6 +26,7 @@ def init_reservations
       .current_in_use
       .merge(Product.alphabetized)
       .includes(:product, order: :user)
+      .sort_by { |r| [r.product.schedule.position ? 0 : 1, r.product.schedule.position] }
```

Comment:
Reservations have no association to Schedule, and belong to Product, not Instrument - so `joins(product: :schedule)` doesn't work here either.

Reply:
The extra logic here is to handle the case of a `nil` position, and sort the `nil`s to the end of the collection.  When the block passed to `sort_by` returns an array, the sort order is based on multiple values (first based on whether the position is present, and then based on the position value).